graphql/README.md

```diff
@@ -0,0 +1,74 @@
+## GraphQL Developer Guide
+
+### Modifying the GraphQL Schema
+
+#### Fields
+
+You can add fields to the .graphql files in the `/schema` folder. When you run `make gqlgen`, these changes will be processed and a resolver will be generated for you.
+
+You can also add models in `gqlgen.yml`. If a GraphQL object has a corresponding model definition in `gqlgen.yml`, then a resolver will not be generated. Instead you may have to edit the API models which are located in the `rest/model` folder.
+
+#### Directives
+
+We have custom directives to control access to certain mutations and queries. They are defined in `schema/directives.graphql` but their corresponding functions are not generated through `make gqlgen`. You will have to manually add or edit the directive functions in `resolver.go`.
+
+### Best Practices for GraphQL
+
+#### Designing Mutations
+
+When designing mutations, the input and payload should be objects. We often have to add new fields, and it is much easier to add backwards compatible changes if the existing fields are nested within an object.
+
+In practice, this means you should prefer
+
+```
+  abortTask(opts: AbortTaskInput!): AbortTaskPayload!
+
+  AbortTaskInput {
+    taskId: String!
+  }
+
+  AbortTaskPayload {
+    task: Task!
+  }
+```
+
+over
+
+```
+  abortTask(taskId: String!): Task!
+```
```

Comment:
I think this should work for syntax highlighting (it looks a little weird here due to the nested triple backticks in the suggestion though)
```suggestion
```graphql
abortTask(opts: AbortTaskInput!): AbortTaskPayload!

AbortTaskInput {
  taskId: String!
}

AbortTaskPayload {
  task: Task!
}
```

over

```Graphql
abortTask(taskId: String!): Task!
```
```